src/ngraph_builder.cc

```diff
@@ -446,6 +446,62 @@ static Status TranslateAddNOp(
   return Status::OK();
 }
 
+static Status TranslateAllOp(const Node* op,
+                             const std::vector<const Tensor*>& static_input_map,
+                             Builder::OpMap& ng_op_map) {
+  shared_ptr<ng::Node> ng_input, ng_axes_op;
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input, &ng_axes_op));
+
```

Comment:
We can add a check for invalid data type for input, since it can have only DT.BOOL. It would be also interesting to see, where the op fails(if it fails at all) if we pass input tensor of data type other than bool. 

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/ops.pbtxt

Reply:
After talking to @sayantan-nervana my understanding is that tf will throw an error in case of an invalid datatype and it will not reach nGraph.

Reply:
If we can confirm TF is doing this check for us, we can close this. Sometimes the TF checks are done in kernel, in which case we might not reach the check.